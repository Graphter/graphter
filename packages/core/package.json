{
  "name": "@graphter/core",
  "version": "1.0.0",
  "description": "",
  "main": "dist/index.js",
  "module": "dist/index.es.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "scripts": {
    "test": "jest --config jest.config.js",
    "test:watch": "react-scripts test --env=jsdom",
    "test:update-snapshots": "jest --updateSnapshot",
    "build": "rollup -c",
    "start": "rollup -c -w",
    "prepare": "npm run build"
  },
  "author": "Jonathan Pritchard",
  "license": "ISC",
  "peerDependencies": {
    "react": "^16.13.1",
    "recoil": "0.0.8"
  },
  "dependencies": {},
  "devDependencies": {
    "@babel/core": "^7.9.0",
    "@babel/plugin-proposal-export-default-from": "^7.8.3",
    "@babel/preset-env": "^7.9.0",
    "@babel/preset-stage-0": "^7.8.3",
    "@rollup/plugin-json": "^4.0.3",
    "@types/jest": "^25.2.1",
    "@types/react": "^16.9.49",
    "@types/recoil": "0.0.1",
    "ajv": "^6.12.2",
    "eslint": "^6.8.0",
    "jest": "^25.5.4",
    "rollup": "^2.2.0",
    "rollup-plugin-babel": "^4.4.0",
    "rollup-plugin-commonjs": "^10.1.0",
    "rollup-plugin-json": "^4.0.0",
    "rollup-plugin-node-resolve": "^5.2.0",
    "rollup-plugin-peer-deps-external": "^2.2.2",
    "rollup-plugin-typescript2": "^0.26.0",
    "ts-jest": "^25.3.1",
    "typescript": "^3.8.3"
  }
}
