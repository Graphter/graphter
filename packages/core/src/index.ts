export * from './ListDisplayConfig'
export * from './Service'
export * from './ListResult'
export * from './GetResult'
export * from './SaveResult'
export * from './NodeConfig'
export * from './NodeRendererProps'
export * from './ErrorRendererProps'
export * from './NodeRendererRegistration'
export * from './JsonType'
export * from './PathSegment'
export * from './NodeValidation'
export * from './ValidationErrorDisplalyMode'
export * from './ValidationExecutionStage'
export * from './ValidationResult'
export * from './NodeValidatorRegistration'