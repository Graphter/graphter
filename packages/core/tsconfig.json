{
  "compilerOptions": {
    "declaration": true,
    "declarationDir": "./dist",
    "module": "esnext",
    "outDir": "./dist",
    "target": "es5",
    "jsx": "react",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "allowSyntheticDefaultImports": true,
    "moduleResolution": "node",
    "allowJs": true,
    "sourceMap": true,
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "skipLibCheck": true,
    "esModuleInterop": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules"
  ]
}
