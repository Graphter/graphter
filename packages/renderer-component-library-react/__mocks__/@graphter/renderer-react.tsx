import React, { useState } from "react";

export const useArrayNodeData = jest.fn(() => {
  return {
    childIds: [],
    removeItem: jest.fn(),
    commitItem: jest.fn()
  }
})

export const useNodeData = jest.fn(() => {
  return [
    null,
    () => {}
  ]
})

export const useNodeValidation = jest.fn().mockResolvedValue([])

export const nodeRendererStore = {
  get: jest.fn((nodeType: string) => ({
    type: 'string',
    renderer: ({originalNodeData}: any) => {
      const [ value, setValue ] = useState(originalNodeData || '')
      return (
        <input value={value} onChange={e => setValue(e.target.value)} data-testid='string-renderer' />
    )
    }
  }))
}
export const createDefault = jest.fn()