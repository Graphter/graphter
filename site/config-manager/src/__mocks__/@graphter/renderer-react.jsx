"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pathConfigStore = exports.createDefault = exports.nodeRendererStore = exports.useNodeValidation = exports.useNodeData = exports.useArrayNodeData = void 0;
var react_1 = require("react");
exports.useArrayNodeData = jest.fn(function () {
    return {
        childIds: [],
        removeItem: jest.fn(),
        commitItem: jest.fn()
    };
});
exports.useNodeData = jest.fn(function () {
    return [
        null,
        function () { }
    ];
});
exports.useNodeValidation = jest.fn().mockResolvedValue([]);
exports.nodeRendererStore = {
    get: jest.fn(function (nodeType) { return ({
        type: 'string',
        renderer: function (_a) {
            var originalNodeData = _a.originalNodeData;
            var _b = react_1.useState(originalNodeData || ''), value = _b[0], setValue = _b[1];
            return (<input value={value} onChange={function (e) { return setValue(e.target.value); }} data-testid='string-renderer'/>);
        }
    }); })
};
exports.createDefault = jest.fn();
exports.pathConfigStore = {
    get: jest.fn(),
    set: jest.fn()
};
