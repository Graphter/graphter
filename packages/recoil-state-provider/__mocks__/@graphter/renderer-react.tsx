import { NodeRendererRegistration } from "@graphter/core";
import React from "react";

export const nodeRendererStore = {
  get: jest.fn()
}

export const pathConfigStore = {
  get: jest.fn(),
  set: jest.fn()
}

export const NodeValidationProvider = jest.fn(({ children }: any) => <div>{children}</div>)
export const NodeDataProvider = jest.fn(({ children }: any) => <div>{children}</div>)