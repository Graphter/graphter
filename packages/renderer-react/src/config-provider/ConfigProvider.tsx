import React, {createContext, useContext} from 'react';
import { NodeConfig, PathSegment } from "@graphter/core";

interface ConfigProviderProps {
  configs: Array<NodeConfig>
  children: any
}

const Context = createContext<Map<PathSegment, NodeConfig> | null>(null);

export function useConfig(id: string): NodeConfig {
  const configMap = useContext(Context)
  if(!configMap) throw new Error(`Couldn't find the config registry. Make sure you've declared a <ConfigProvider /> and passed it valid config.`)
  const config = configMap.get(id)
  if(!config) throw new Error(`Missing '${id}' config`)
  return config
}

export default function ConfigProvider({ configs, children }: ConfigProviderProps){
  const configMap = configs.reduce((a, c) => {
    if(Array.isArray(c)) c.forEach(registration => a.set(registration.id, registration.config))
    else a.set(c.id, c)
    return a
  }, new Map<string, NodeConfig>())
  return (
    <Context.Provider value={configMap}>
      {children}
    </Context.Provider>
  );
}