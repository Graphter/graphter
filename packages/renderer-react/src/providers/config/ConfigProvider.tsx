import React, {createContext, useContext} from 'react';
import { NodeConfig, PathSegment } from "@graphter/core";

interface ConfigProviderProps {
  configs: Array<NodeConfig>
  children: any
}

const Context = createContext<Map<string | number, NodeConfig> | null>(null);

let configMap: Map<string | number, NodeConfig>

export function getConfig(id: string | number): NodeConfig {
  if(!configMap) throw new Error(`Couldn't find the API service registry. Make sure you've declared a <ServiceProvider /> and passed it a valid service.`)
  const service = configMap.get(id)
  if(!service) throw new Error(`Missing '${id}' config`)
  return service
}

export function useConfig(id: string | number): NodeConfig {
  const configMap = useContext(Context)
  if(!configMap) throw new Error(`Couldn't find the config registry. Make sure you've declared a <ConfigProvider /> and passed it valid config.`)
  const config = configMap.get(id)
  if(!config) throw new Error(`Missing '${id}' config`)
  return config
}

export default function ConfigProvider({ configs, children }: ConfigProviderProps){
  configMap = configs.reduce((a, c) => {
    if(Array.isArray(c)) c.forEach(registration => a.set(registration.id, registration.config))
    else a.set(c.id, c)
    return a
  }, new Map<string, NodeConfig>())

  return (
    <Context.Provider value={configMap}>
      {children}
    </Context.Provider>
  );
}