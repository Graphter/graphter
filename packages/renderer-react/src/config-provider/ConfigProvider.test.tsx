import React from 'react';
import ConfigProvider, {useConfig} from "./ConfigProvider";
import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom'
import { NodeConfig } from "@graphter/core";

describe(`<ConfigProvider />`, () => {
  let mockConfig: NodeConfig;

  function ConsumerMock({ cb }: { cb?: (config: NodeConfig) => void }) {
    const config = useConfig('mock-config');
    cb && cb(config)
    return null;
  }

  beforeEach(() => {
    mockConfig = {
      id: 'mock-config',
      type: 'mock'
    };
  });

  it(`should render children`, () => {
    render(
      <ConfigProvider configs={[
        mockConfig
      ]}>
        <div data-testid='some-child-component'>Some child component</div>
      </ConfigProvider>
    );
    const childComponent = screen.queryByTestId('some-child-component');
    expect(childComponent).not.toBeNull();
  });

  describe(' when using useConfig', () => {

    it(`should error when no provider is declared`, () => {
      expect(() => (
        render(<ConsumerMock />)
      )).toThrowErrorMatchingSnapshot();
    });

    it(`should return the config passed to the provider`, () => {
      const cbFn = jest.fn()
      render(
        <ConfigProvider configs={[
          mockConfig
        ]}>
          <ConsumerMock cb={cbFn}/>
        </ConfigProvider>
      );
      expect(cbFn).toHaveBeenCalledWith(mockConfig)
    });

  });

});


export {};