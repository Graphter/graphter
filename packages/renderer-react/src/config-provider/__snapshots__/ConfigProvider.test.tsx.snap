// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigProvider />  when using useConfig should error when no provider is declared 1`] = `"Couldn't find the config registry. Make sure you've declared a <ConfigProvider /> and passed it valid config."`;
